### PR TITLE
Add support for switch expressions to Resource Leak Checker

### LINKED ISSUE #4993: Support switch expressions in the Resource Leak Checker
This will require some (hopefully modest) modifications to the `MustCallConsistencyAnalyzer`.  Fortunately, there is already a synthetic variable for the value of each `switch` expression created during CFG construction.  So I think all we will have to do is treat that variable as the "temporary" representing the `switch` expression, and everything else should work out.

### PR DESCRIPTION
**Pull request by msridhar**
The only change required was to update temporary variable mappings, like we now do for ternary expressions.

Fixes #4993

### CODE REVIEW

File: checker/tests/resourceleak/SwitchExpressions.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
+// @below-java17-jdk-skip-test
```
Comment (msridhar) [inline checker/tests/resourceleak/SwitchExpressions.java:1]: These tests are exactly the same as those in `TernaryExpressions`, but using switch expressions instead

**Comment by msridhar** [comment]
FYI @Nargeshdb

**Review by kelloggm — DISMISSED** [review]
LGTM

**Comment by msridhar** [comment]
Sorry @kelloggm your review got dismissed because I just moved a file to a different directory

**Review by kelloggm — APPROVED** [review]

**Pull request merged**